Source/WebInspectorUI/UserInterface/Views/ObjectTreePropertyTreeElement.js

```diff
@@ -159,8 +159,26 @@ WI.ObjectTreePropertyTreeElement = class ObjectTreePropertyTreeElement extends W
         if (this._mode === WI.ObjectTreeView.Mode.Properties) {
             if (this.property.isPrivateProperty)
                 nameElement.classList.add("private");
-            if (!this.property.enumerable)
-                nameElement.classList.add("not-enumerable");
+            if (this.property.isInternalProperty)
+                nameElement.classList.add("internal");
+
+            nameElement.title += "\n\n" + [
+                (() => {
+                    if (this.property.configurable)
+                        return WI.UIString("Configurable", "Configurable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is configurable.");
+                    return WI.UIString("Not configurable", "Not configurable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is not configurable.");
+                })(),
+                (() => {
+                    if (this.property.enumerable)
+                        return WI.UIString("Enumerable", "Enumerable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is enumerable.");
+                    return WI.UIString("Not enumerable", "Not enumerable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is not enumerable.");
+                })(),
+                (() => {
+                    if (this.property.writable)
+                        return WI.UIString("Writable", "Writable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is writable.");
+                    return WI.UIString("Not writable", "Not writable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is not writable.");
+                })(),
+            ].join("\n");
```

Comment:
Is the array join getting us something here that some if/else statements appending directly to the title wouldn't? Seems like we could just 
```
if (this.property.configurable)
    nameElement.title += "\n" + WI.UIString("Configurable", "Configurable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is configurable.");
else
    nameElement.title += "\n" + WI.UIString("Not configurable", "Not configurable @ Object Tree Property", "Part of the tooltip indicating that the hovered property is not configurable.");
```
and so on. Not opposed, but curious if there is some benefit to your approach I'm missing because I find it harder to read at a glance than just some if statements appending to the title. (and either way we end up having to append each element and each newline to the string…)

Reply:
my original hope was to avoid modifying `nameElement.title` a bunch of times as that probably wouldve resulting in a a bunch of copies, but i think I agree that it'll look nicer to read :)